README.md

```diff
@@ -1,4 +1,172 @@
+</table>
+
+## Tree2Seq framework
+
+Implementation based on [PyTorch](https://pytorch.org/docs/stable/torch.html) framework.
+For supporting learning on graphs [DGL](https://www.dgl.ai/) library is used.
+
+Framework consist of 3 main parts:
+- Node embedding (`INodeEmbedding`) that embed tokens in nodes to vectors.
+- Tree encoder (`ITreeEncoder`) that encode tree into a vector representation.
+- Tree decoder (`ITreeDecoder`) that decode vector representation into target sequence.
+
+Converting source tree into target sequence occurs in `Tree2Seq` class and represent sequence applying of interface realization.
+If you want to add a new algorithm, you should realize class with inheritance from the needed interface, and then use it in the configuration.
+
+### Configuration
+
+Components configuration and required hyper parameters can be passed to the model throw `.json` files.
+The examples of such files can be found in `config` directory, there are already prepared configs for training and evaluating
+Tree-LSTM model on Java dataset.
+
+Use can use prepared scripts to interact with model:
+- Start training ChildSum Tree-LSTM with logging to [wandb](https://www.wandb.com/) service
+```bash
+python train.py configs/tree_lstm_childsum_java_small.json wandb
+```
+- Evaluate model:
+```bash
+python evaluate.py configs/evaluate.json
+```
+- Predict target sequence for given file with concrete model (predict function name by its body),
+this step includes data preprocessing and model applying:
+```bash
+python interactive.py test.java model.pt
+```
+
+## Evaluation
+
+Currently, there are implemented Tree-LSTM model with various modifications. For all of them, LSTM with Attention to subtrees
+is used to decoder to target sequence (method name). For node embedding used type (internal nodes) and token (leafs) matrix embeddings.
+
+<table>
+<thead>
+  <tr>
+    <th rowspan="2">Model</th>
+    <th colspan="3">Java-small</th>
+    <th colspan="3">Java-medium</th>
+  </tr>
+  <tr>
+    <td>Precision</td>
+    <td>Recall</td>
+    <td>F1-score</td>
+    <td>Precision</td>
+    <td>Recall</td>
+    <td>F1-score</td>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>ChildSum Tree-LSTM</td>
+    <td>47.12</td>
+    <td>30.27</td>
+    <td>36.86</td>
+    <td>56.11</td>
+    <td>38.02</td>
+    <td>45.33</td>
+  </tr>
+  <tr>
+    <td>Attn. Tree-LSTM</td>
+    <td>48.07</td>
+    <td>30.44</td>
+    <td>37.27</td>
+    <td>56.52</td>
+    <td>38.16</td>
+    <td>45.53</td>
+  </tr>
+  <tr>
+    <td>Sequence Tree-LSTM</td>
+    <td>47.65</td>
+    <td>30.53</td>
+    <td>37.21</td>
+    <td>56.84</td>
+    <td>38.70</td>
+    <td>46.05</td>
+  </tr>
+  <tr>
+    <td>Conv. Tree-LSTM</td>
+    <td>47.29</td>
+    <td>30.06</td>
+    <td>36.76</td>
+    <td>56.97</td>
+    <td>38.86</td>
+    <td>46.20</td>
+  </tr>
+</tbody>
+</table>
+
+## Contribution
+
+Supporting different algorithms to encode and decode trees are the main key areas of improvement this framework.
```

Comment:
Supporting different algorithms of encoding and decoding trees is the key area of improvement for this framework.
If you have any suggestions or questions, feel free to open issues and create pull requests